ppb/fields.py

```diff
@@ -0,0 +1,91 @@
+"""
+Field declaration system.
+
+This allows applying coercion and other code to both instances and subclasses.
+
+class Spam(FieldMixin):
+    class Fields:
+        eggs: str
+        foo = BarProperty()
+"""
+
+__all__ = 'FieldMixin', 'iterfields',  # 'field', 'virtualfield'
+
+
+def _annotations_to_fields(annos):
+    """
+    Converts an annotations dict into a fields dict
+    """
+
+
+class FieldMixin:
+    """
+    Mixin that implements all the field magic
+    """
+    def __init_subclass__(cls, **_):
+        # If new fields are defined, make a __fields__
```

Comment:
Should we worry about the lack of super call?

Reply:
I probably just forgot about it.